Do not clear content filter when loading after serviceWorkerDidNotHandle

pascoej opened the pull request:
#### 36a507b11f956b34bdaccd7785a4c484f90befe5
<pre>
Do not clear content filter when loading after serviceWorkerDidNotHandle
<a href="https://bugs.webkit.org/show_bug.cgi?id=257569">https://bugs.webkit.org/show_bug.cgi?id=257569</a>
rdar://109619110

Reviewed by Youenn Fablet.

The content filter is cleared after serviceWorkerDidNotHandle is called, but another
request can be made using the same NetworkResourceLoader, without a content filter.

This patch fixes that by not clearing the filter.

* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
(WebKit::NetworkResourceLoader::serviceWorkerDidNotHandle):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ContentFiltering.mm:
(TEST):

Canonical link: <a href="https://commits.webkit.org/265022@main">https://commits.webkit.org/265022@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aac5f01d9620d2b8197c4450e375912e8b75d9bc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/9396 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9673 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/9904 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/11058 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/9268 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/9405 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11660 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9635 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/12157 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/9547 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/10461 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8075 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/11215 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/7758 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/8577 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/16006 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/8855 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/8726 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12053 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9219 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/7502 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/8431 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2293 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/12656 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/8981 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/ContentFiltering.mm:
```diff
@@ -146,6 +193,80 @@ - (void)webView:(WKWebView *)webView didCommitNavigation:(WKNavigation *)navigat
+    "{"
+    "    frame = document.createElement('iframe');"
+    "    frame.src = \"/test.html\";"
+    "    frame.onload = function() { window.webkit.messageHandlers.sw.postMessage(frame.contentDocument.body.innerHTML); }"
```
Could use log instead

youennf replied:
Could use log instead

youennf commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/ContentFiltering.mm:
```diff
@@ -108,6 +109,52 @@ - (void)encodeWithCoder:(NSCoder *)coder
+
+@end
+
+static void loadAlternateTest(Decision decision, DecisionPoint decisionPoint)
```
Is it necessary to move this function?

pascoej replied:
In order to make the test able to find all the functions where I initially placed it, yes. But I think I can just add the new test to the bottom to avoid this. I will make that change as not to pollute the blame too much.

youennf commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/ContentFiltering.mm:
```diff
@@ -108,6 +109,52 @@ - (void)encodeWithCoder:(NSCoder *)coder
+    [webView evaluateJavaScript:@"document.body.innerText" completionHandler:^(id result, NSError *error) {
+        EXPECT_TRUE([result isKindOfClass:[NSString class]]);
+        EXPECT_WK_STREQ(@"blocked", result);
+        isDone = true;
```
I know this is existing code but having EXPECT checks here makes the test a bit hard to read.
The test itself has no EXPECT checks, it feels like magic.
It could be easier to read if we would wait until done is true in the test, then do a EXPECT_TRUE... in the test itself.

youennf commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/ContentFiltering.mm:
```diff
@@ -146,6 +193,80 @@ - (void)webView:(WKWebView *)webView didCommitNavigation:(WKNavigation *)navigat
+
+        TestWebKitAPI::HTTPServer server({
+            { "/"_s, { mainForFetchTestBytes } },
+            { "/sw.js"_s, { { { "Content-Type"_s, "application/javascript"_s } }, serviceWorkerJS } },
```
We could add a "/test.html" entry for good measure.

pascoej committed:
Do not clear content filter when loading after serviceWorkerDidNotHandle
https://bugs.webkit.org/show_bug.cgi?id=257569
rdar://109619110

Reviewed by Youenn Fablet.

The content filter is cleared after serviceWorkerDidNotHandle is called, but another
request can be made using the same NetworkResourceLoader, without a content filter.

This patch fixes that by not clearing the filter.

* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
(WebKit::NetworkResourceLoader::serviceWorkerDidNotHandle):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ContentFiltering.mm:
(TEST):

Canonical link: https://commits.webkit.org/265022@main
